packages/expo-modules-autolinking/scripts/ios/project_integrator.rb

```diff
@@ -128,5 +132,124 @@ def self.set_autolinking_configuration(project)
+        puts "[Expo] ".blue << "Installing the build script for target " << native_target.name.green
+
+        # Create a new build script in the target, it's added as the last phase
+        xcode_build_script = native_target.new_shell_script_build_phase(Constants::CONFIGURE_PROJECT_BUILD_SCRIPT_NAME)
+      end
+
+      # Make sure it is before the "Compile Sources" build phase
+      xcode_build_script_index = build_phases.find_index(xcode_build_script)
+      compile_sources_index = build_phases.find_index { |phase|
+        phase.is_a?(Xcodeproj::Project::PBXSourcesBuildPhase)
+      }
+
+      # Insert our script before the "Compile Sources" phase when necessary
+      unless compile_sources_index.nil? || xcode_build_script_index < compile_sources_index
+        build_phases.insert(
+          compile_sources_index,
+          build_phases.delete_at(xcode_build_script_index)
+        )
+      end
+
+      # Get path to the script that will be added to the target support files
+      support_script_path = File.join(target.support_files_dir, Constants::CONFIGURE_PROJECT_SCRIPT_FILE_NAME)
+      support_script_relative_path = Pathname.new(support_script_path).relative_path_from(project.project_dir)
+
+      # Write to the shell script so it's always in-sync with the autolinking configuration
+      IO.write(
+        support_script_path,
+        generate_support_script(autolinking_manager, modules_provider_path)
+      )
+
+      # Make the support script executable
+      FileUtils.chmod('+x', support_script_path)
+
+      # Force the build phase script to run on each build (including incremental builds)
+      xcode_build_script.always_out_of_date = '1'
+
+      # Make sure the build script in Xcode is up to date, but probably it's not going to change
+      # as it just runs the script generated in the target support files
+      xcode_build_script.shell_script = generate_xcode_build_script(support_script_relative_path)
+    end
+
+    # Generates the shell script of the build script phase.
+    # Try not to modify this since it involves changes in the pbxproj so
+    # it's better to modify the support script instead, if possible.
+    def self.generate_xcode_build_script(script_relative_path)
+      escaped_path = script_relative_path.to_s.gsub(/[^a-zA-Z0-9,\._\+@%\/\-]/) { |char| "\\#{char}" }
+
+      <<~XCODE_BUILD_SCRIPT
+      # This script configures Expo modules and generates the modules provider file.
+      bash -l -c "./#{escaped_path}"
+      XCODE_BUILD_SCRIPT
+    end
+
+    # Generates the support script that is executed by the build script phase.
+    def self.generate_support_script(autolinking_manager, modules_provider_path)
+      args = autolinking_manager.base_command_args.map { |arg| "\"#{arg}\"" }.join(' ')
+
+      <<~SUPPORT_SCRIPT
+      #!/usr/bin/env bash
+      # @generated by expo-modules-autolinking
+
+      set -eo pipefail
+
+      function node() {
```

Comment:
```suggestion
      function with_node() {
```

prefer rename to with_node, see context for the next comment